pkg/syncer/mutators/interface.go

```diff
@@ -0,0 +1,26 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mutators
+
+import (
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+type Mutator interface {
+	ApplySpec(downstreamObj *unstructured.Unstructured) error
```

Comment:
If spec and status mutation are orthogonal, might it make sense to just use a function (e.g. `mutate(*unstructured.Unstructured) error`) and define mutators separately for spec and status? 

That would simplify testing - there wouldn't have to be special-case handling of spec vs status. The effective type of a mutator would be determined by which syncer (spec or status) it was registered to. They could still be defined together, but having a mutator for one type of syncer wouldn't require an implementation (even if empty) for the other.